src/pages-helpers/curriculum/docx/builder/helper.ts

```diff
@@ -0,0 +1,163 @@
+import { sum } from "lodash";
+import JSZip from "jszip";
+
+import { Slugs } from "..";
+import { zipToSimpleObject } from "../zip";
+
+import {
+  Thread,
+  Unit,
+} from "@/components/CurriculumComponents/CurriculumVisualiser";
+
+/**
+ * Uncapitalize everything except acronyms
+ * @param input capitalized string
+ * @returns uncapitalized string
+ */
+export function uncapitalize(input: string, titleCaseWords: string[] = []) {
+  const chars = input.split("");
+  let output = ``;
+  for (let i = 0; i < chars.length; i++) {
+    let j = i;
+    let allUpperCase = true;
+    let doBreak = false;
+    let word = "";
+    while (j < chars.length) {
+      if (chars[j] === " ") {
+        doBreak = true;
+      } else if (!chars[j]!.match(/[A-Z]/)) {
+        allUpperCase = false;
+      }
+      word += chars[j];
+      j++;
+      if (doBreak) break;
+    }
+
+    if (titleCaseWords.includes(word.toLowerCase())) {
+      const firstLetter = word.slice(0, 1);
+      const rest = word.slice(1);
+      output += `${firstLetter.toUpperCase()}${rest}`;
+    } else if (allUpperCase) {
+      output += word;
+    } else {
+      output += word.toLowerCase();
+    }
+    i = j - 1;
+  }
+  return output;
+}
+
+/**
+ * Uncapitalize with language exceptions
+ * @param input capitalized string
+ * @returns uncapitalized string
+ */
+export function uncapitalizeSubject(input: string) {
+  return uncapitalize(input, ["english", "french", "spanish", "german"]);
+}
+
+export function notUndefined<TValue>(
+  value: TValue | undefined,
+): value is TValue {
+  return value !== undefined;
+}
+
+export function createCurriculumSlug(slugs: Slugs) {
+  return `${slugs.subjectSlug}-${slugs.phaseSlug}${
+    slugs.examboardSlug ? `-${slugs.examboardSlug}` : ""
+  }`;
+}
+
+// TODO: This is from from Sonali's work that's not yet merged
```

Comment:
This is dup code and should be removed and use the in existing version instead

Reply:
Fixed in https://github.com/oaknational/Oak-Web-Application/pull/2552/commits/b054774714824e83b2c202da309afc89027745dc